src/main/java/seedu/taskify/logic/parser/ParserUtil.java

```diff
@@ -34,9 +43,52 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
         if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
             throw new ParseException(MESSAGE_INVALID_INDEX);
         }
+
         return Index.fromOneBased(Integer.parseInt(trimmedIndex));
     }
 
+    /**
+     * Parses {@code oneBasedIndexes} into a list of {@code Index} and returns it.
+     * @param oneBasedIndexes user's input excluding the command word
+     * @return a list of {@code Index} representing all valid indexes in {@code oneBasedIndexes}
+     * @throws ParseException if {@code oneBasedIndexes} cannot be parsed properly
+     */
+    public static List<Index> parseMultipleIndex(String oneBasedIndexes) throws ParseException {
+        assert hasMultipleValidIndex(oneBasedIndexes) : "ParserUtil#parseMultipleIndex should not be called on this "
```

Comment:
Should an exception be used here instead?

Reply:
ParseExceptions are thrown in line 59 and line 64 for unexpected user input, my assertion in line 57 was mainly to warn whoever make changes to our code against using this public method. It should only be called if the user's argument input passes `#hasMultipleValidIndex` first, as seen in line 31 of `DeleteMultipleCommandParser` and line 59 of `TaskifyParser`.

Reply:
Ok makes sense